working-with-fhir-igs/hl7v2-to-dk-medcom/utils.bal

```diff
@@ -0,0 +1,144 @@
+            r4:Resource mergedResource = check merged.cloneWithType(r4:Resource);
+            updatedEntries.push({'resource: mergedResource});
+        }
+    }
+    bundle.entry = updatedEntries;
+
+    return bundle;
+
+}
+
+// Utility function to merge JSON objects with precedence for first object
+
+public isolated function deepMergeJson(json firstJson, json secondJson) returns json|error {
+    // Handle non-object JSON types
+    if (!(firstJson is map<json>) || !(secondJson is map<json>)) {
+        // If first is not an object or both are not objects, return first (precedence)
+        if (!(firstJson is map<json>)) {
+            return firstJson;
+        }
+        // If only second is not an object, return first
+        return firstJson;
+    }
+
+    map<json> result = {};
+    map<json> secondMap = <map<json>>secondJson;
+    map<json> firstMap = <map<json>>firstJson;
+
+    // First, add all fields from second JSON
+    foreach var [key, val] in secondMap.entries() {
+        // Skip empty arrays and empty strings
+        if (val is string && val == "") {
+            continue;
+        } else if (val is json[] && val.length() == 0) {
+            continue;
+        } else {
+            result[key] = val;
+        }
+    }
+
+    // Then process all fields from first JSON (taking precedence)
+    foreach var [key, val] in firstMap.entries() {
+        // Skip empty arrays and empty strings
+        if (val is string && val == "") {
+            continue;
+        } else if (val is json[] && val.length() == 0) {
+            continue;
+        }
+        // Deep merge for nested objects
+        else if (val is map<json> && result.hasKey(key) && result[key] is map<json>) {
+            result[key] = check deepMergeJson(val, result[key]);
+        }
+        // Array concatenation - combine arrays from both objects
+        else if (val is json[] && result.hasKey(key) && result[key] is json[]) {
+            json[] firstArray = <json[]>val;
+            json[] secondArray = <json[]>result[key];
+
+            // Single element arrays will be treated as special case
+            if (firstArray.length() == 1 && secondArray.length() == 1) {
+                // If both arrays have a single element, merge them
+                if (val[0] is map<json> && secondArray[0] is map<json>) {
+                    result[key] = [check deepMergeJson(val[0], secondArray[0])];
+                } else {
+                    // If both arrays have a single element, keep the first one
+                    result[key] = [firstArray[0]];
+                }
+                continue;
+            }
+
+            // Concatenate arrays, putting first object's items first
+            result[key] = [...firstArray, ...secondArray];
+        }
+        // For all other cases, first object takes precedence
+        else {
+            result[key] = val;
+        }
+    }
+
+    return result;
+}
```

Comment:
Let's move this as a utility given by r4

Reply:
ack. created an issue: https://github.com/wso2-enterprise/open-healthcare/issues/1724